### PR TITLE
profile logic

### PR DESCRIPTION
**Pull request by TijanAyo**
The profile page logic

### CODE REVIEW

File: src/profile/profile.controller.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+export class ProfileController {
+    constructor(private profileService: ProfileService) {}
+
+    @Post('/create')
+    async createProfile(@Body() body: CreateProfileDto) {
+        const userProfile = await this.profileService.createprofile(body)
+        return userProfile;
+    }
+
+    @Get('/:id')
```
Comment (Chimexx) [inline src/profile/profile.controller.ts:15]: image

This is how your controllers should look like, Compare the differences .

**Review by Chimexx — CHANGES_REQUESTED** [review]
I can't find any migration file, which means you didn't test this code, This is the second time you are doing this, I no understand you oh.

Now how do we know one can successfully create and save a user in the db?

**Comment by TijanAyo** [comment]
I didn't run migration

But I tested it... when I run start:dev the table are created on the db automatically

File: src/profile/profile.controller.ts (hunk 1 above)
Comment (TijanAyo) [reply src/profile/profile.controller.ts:15]: You have to take my word that is working😅😅
 ... currently having a hard time migrating, the migration just keeps pushing a migration table to the DB and not the entities table.

And for the second one... You mentioned in one of my last pushes that you don't use **ZaLaResponse** in the controller hence why I didn't add that

Comment (SunmadeBabafemi) [reply src/profile/profile.controller.ts:15]: you are misunderstanding him.. don't use ZaLaResponse in services but controllers.. you only use ZaLaResponse in services when you are trying to catch errors

Comment (TijanAyo) [reply src/profile/profile.controller.ts:15]: Ohhh I think I get what you guys are trying to say... I will fix that now 

Thank you

Comment (TijanAyo) [reply src/profile/profile.controller.ts:15]: What I get any time I run a migration

image

And this is what appears in my DB

image

Comment (Chimexx) [reply src/profile/profile.controller.ts:15]: If you have an issue, you ask for help...Don't go ahead to push code that  is potentially buggy.

Comment (TijanAyo) [reply src/profile/profile.controller.ts:15]: Help ... 
🦸🏾‍♂️

I can't seem to run migration

Comment (Chimexx) [reply src/profile/profile.controller.ts:15]: Add migration file with: npm run typeorm migration:generate ./src/database/migration/${file name}

Then: npm run migration

But before then ensure your env and db is properly setup

Comment (SunmadeBabafemi) [reply src/profile/profile.controller.ts:15]: exactly... change the ${file name} to the name of the migration file you want to generate
...

Comment (TijanAyo) [reply src/profile/profile.controller.ts:15]: Thank you, guys... y'all just saved me lots of headaches

File: src/profile/dto/create-profile.dto.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export class CreateProfileDto {
+    @IsString()
+    @IsNotEmpty()
+    @ApiProperty()
+    email: string
+
+    @IsUUID()
+    @IsNotEmpty()
+    @ApiProperty()
+    userID: string
```
Comment (PhilemonBrain) [inline src/profile/dto/create-profile.dto.ts:13]: userId not userID please

File: src/entities/profile.entity.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { SharedEntity } from "../common/model/sharedEntity";
+import { Column, Entity } from "typeorm";
+
+@Entity()
+export class Profile extends SharedEntity {
+    @Column()
+    email: string;
+
+    @Column()
+    userID: string
```
Comment (PhilemonBrain) [inline src/entities/profile.entity.ts:10]: userId not userID

File: src/profile/profile.service.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        catch(err){
+            throw new BadRequestException(
+                ZaLaResponse.BadRequest('Internal Server Error', err.message, '500')
+            )
+        }
+    }
+
+   async getone(profileID: string):Promise<Profile>{
+    try{
+        const profile = await this.profileRepo.findOneByOrFail({
```
Comment (PhilemonBrain) [inline src/profile/profile.service.ts:41]: findOneByOrFail throws an error, dosent return undefined

File: src/profile/profile.service.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            return newProfile;
+        }
+        catch(err){
+            throw new BadRequestException(
+                ZaLaResponse.BadRequest('Internal Server Error', err.message, '500')
+            )
+        }
+    }
+
+   async getone(profileID: string):Promise<Profile>{
```
Comment (PhilemonBrain) [inline src/profile/profile.service.ts:39]: profileId not profileId

**Review by PhilemonBrain — CHANGES_REQUESTED** [review]

**Review by Chimexx — APPROVED** [review]

**Comment by Chimexx** [comment]
@TijanAyo Approved, just resolve your conflicts

**Review by SunmadeBabafemi — APPROVED** [review]

**Pull request merged**